tools/lsp/common.rs

```diff
@@ -16,6 +17,31 @@ pub type UrlVersion = Option<i32>;
 #[cfg(target_arch = "wasm32")]
 use crate::wasm_prelude::*;
 
+pub fn extract_element(node: SyntaxNode) -> Option<syntax_nodes::Element> {
+    match node.kind() {
+        SyntaxKind::Element => Some(node.into()),
+        SyntaxKind::SubElement => extract_element(node.child_node(SyntaxKind::Element)?),
+        SyntaxKind::ConditionalElement | SyntaxKind::RepeatedElement => {
+            extract_element(node.child_node(SyntaxKind::SubElement)?)
+        }
+        _ => None,
+    }
+}
+
+/// Use this in nodes you want the language server and preview to
+/// ignore a node for code analysis purposes.
+pub const NODE_IGNORE_COMMENT: &str = "@lsp:ignore-node";
```

Comment:
This is still only useful for the preview side, isn't it?

Reply:
There is even a warning about it in the CI. I suggest to put it back in the preview module